source/gameengine/Rasterizer/RAS_OpenGLRasterizer/RAS_StorageVA.cpp

```diff
@@ -186,8 +187,43 @@ void RAS_StorageVA::IndexPrimitives(RAS_MeshSlot *ms)
 	else
 		glColor4f(0.0f, 0.0f, 0.0f, 1.0f);
 
-	// here the actual drawing takes places
-	glDrawElements(array->GetOpenGLPrimitiveType(), array->m_index.size(), GL_UNSIGNED_INT, indexarray);
+	/* For big display lists, using only one draw call with glDrawElements can generate a bottleneck
+	 * in Vertex Array mode, and cause a performance decrease. So we split the array list into
+	 * several parts and make one draw call for each part with glDrawRageElements */
+	if (array->m_index.size() > 50000) {
```

Comment:
Maybe you can add a check if the VA is using DL, if is not the case don't compute parts.